### PR TITLE
Allow custom headers to be set in Server

### PR DESCRIPTION
**Pull request by RohitK89**
Opening a new PR as #599 was accidentally closed.

This one also includes the requested change in 599 before it was closed.

@overcat

### CODE REVIEW

File: stellar_sdk/client/requests_client.py [hunk 1]
```diff
@@ -50,6 +50,7 @@ def __init__(
         backoff_factor: float = DEFAULT_BACKOFF_FACTOR,
         session: Session = None,
         stream_session: Session = None,
+        custom_headers: Optional[Dict[str, str]] = None,
```
Comment (overcat) [inline stellar_sdk/client/requests_client.py:53]: Please add a description of `custom_headers` in the document.

File: stellar_sdk/server.py [hunk 2]
```diff
@@ -39,17 +39,20 @@ class Server(BaseServer):
...
     :param client: Http client used to send the request
+    :param custom_request_headers: Any headers that should be added when making HTTP requests to the
+    client.
     """
 
     def __init__(
         self,
         horizon_url: str = "https://horizon-testnet.stellar.org/",
         client: BaseSyncClient = None,
+        custom_request_headers: Optional[Dict[str, str]] = None,
```
Comment (overcat) [inline stellar_sdk/server.py:50]: please check https://github.com/StellarCN/py-stellar-base/pull/599#discussion_r895083961

**Review by overcat — CHANGES_REQUESTED** [review]

File: stellar_sdk/server.py (hunk 2 above)
Comment (RohitK89) [reply stellar_sdk/server.py:50]: @overcat I certainly understand that perspective, but here's a counterpoint: Right now in my own code, I've basically had to replicate about 50 lines of code from this file to make my own `Session`, just to be able to add the one additional header in my requests. It is an unnecessary burden on users in such a situation and leads to a poor experience. You'll also find that more consumers of this library might need such a feature, as there are more services out there in the market that let you query Horizon via APIs that require authentication.

Basically, the cost of adding this feature is nothing and the downside is placing an unnecessary burden on users. It seems like an easy win to me.

Comment (overcat) [reply stellar_sdk/server.py:50]: Hi @RohitK89, I'm not sure why we need to add 50 lines of code, is the following code not enough to do the job?
```python
from stellar_sdk import Server
from stellar_sdk.client.requests_client import RequestsClient

client = RequestsClient(custom_headers={"key": "value"})
server = Server("https://horizon-testnet.stellar.org", client)
```

I think `custom_request_headers` is part of `client`, so I still don't want to add it here. If we add it here, should we also add other parameters(`pool_size`, `num_retries`, `request_timeout` etc) here?

**Review by overcat — CHANGES_REQUESTED** [review]

File: stellar_sdk/server.py (hunk 2 above)
Comment (overcat) [reply stellar_sdk/server.py:50]: Hi @RohitK89, I'm not sure why we need to add 50 lines of code, is the following code not enough to do the job?
```python
from stellar_sdk import Server
from stellar_sdk.client.requests_client import RequestsClient

client = RequestsClient(custom_headers={"key": "value"})
server = Server("https://horizon-testnet.stellar.org", client)
```

I think custom_request_headers is part of client, so I still don't want to add it here. If we add it here, should we also add other parameters(pool_size, num_retries, request_timeout etc) here?

Comment (RohitK89) [reply stellar_sdk/server.py:50]: oh, sorry, I misunderstood. I see what you mean - yes, I think it's not unreasonable to add the param only in the client and let users instantiate their own clients and pass to the server. I'll update.

**Review by overcat — APPROVED** [review]
Thank you @RohitK89!
